kaiax/supply/impl/checkpoint.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package supply
+
+import (
+	"math/big"
+	"sync/atomic"
+	"time"
+
+	"github.com/kaiachain/kaia/kaiax/supply"
+)
+
+var (
+	checkpointInterval    = uint64(128)
+	accumulateLogInterval = uint64(102400) // Periodic log in accumulateCheckpoint().
+)
+
+// supplyCheckpoint represents the accumulated minted and burnt fee up to a specific block number.
+type supplyCheckpoint struct {
+	Minted   *big.Int
+	BurntFee *big.Int
+}
+
+func (sc *supplyCheckpoint) Copy() *supplyCheckpoint {
+	return &supplyCheckpoint{
+		Minted:   new(big.Int).Set(sc.Minted),
+		BurntFee: new(big.Int).Set(sc.BurntFee),
+	}
+}
+
+func nearestCheckpointInterval(num uint64) uint64 {
+	return num - (num % checkpointInterval)
+}
+
+// loadLastCheckpoint loads the last supply checkpoint from the database.
+func (s *SupplyModule) loadLastCheckpoint() error {
```

Comment:
Perhaps move SupplyModule methods to `init.go`?

Reply:
Let me move it after we're settle with the SupplyCheckpoint naming.

Reply:
Moved into init.go and getter.go according to their usage.